ratis-server/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java

```diff
@@ -410,7 +410,7 @@ static void setRetentionFileNum(RaftProperties properties, int numSnapshotFilesR
     String PREFIX = RaftServerConfigKeys.PREFIX + ".data-stream";
 
     String ASYNC_THREAD_POOL_SIZE_KEY = PREFIX + ".async.thread.pool.size";
-    int ASYNC_THREAD_POOL_SIZE_DEFAULT = 4;
+    int ASYNC_THREAD_POOL_SIZE_DEFAULT = 16;
```

Comment:
Any particular reason why choosing 16 threads as the number? (or why not choose a large number, like 500)?

Reply:
Choose 16 for a value close to the number of cores of a machine; 4 seems too small.  For large value like 500, it might be good for application like Ozone but the value is too large for general applications.